lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1839,13 +1838,6 @@ private function doPersist($entity, array &$visited): void
             case self::STATE_DETACHED:
                 // Can actually not happen right now since we assume STATE_NEW.
                 throw ORMInvalidArgumentException::detachedEntityCannot($entity, 'persisted');
-
-            default:
-                throw new UnexpectedValueException(sprintf(
-                    'Unexpected entity state: %s. %s',
-                    $entityState,
-                    self::objToStr($entity)
-                ));
```

Comment:
I wouldn't remove these `default`s. They're technically dead code, but I'd consider them a defense against a misbehaving `getEntityState()` implementation. Let's baseline or suppress the error.

On 3.0.x, we could turn the `switch` into a `match` and rely on the runtime to error if none of the cases matched.

Reply:
> Let's baseline or suppress the error.

That will only be doable when using Psalm 5, but OK 👍